### DISCUSSION TITLE
Should I append new dialects at the end, or sort them alphabetically?

### DISCUSSION THREAD

**Discussion by dazhuangc** [sequelize/sequelize#17294] [Help & Questions] [answered]
Hi,

When adding new `hana` dialect, I need to add it into the existing dialects list at the correct position.

For example in `test/unit/query-generator/create-table-query.test.ts`,
Here it looks better to append `hana` to the end:
```javascript
  it('produces a query to create a table with integer with references', () => {
    expectsql(
      queryGenerator.createTableQuery('myTable', { myColumn: 'INTEGER REFERENCES "Bar" ("id")' }),
      {
        default: ...,
        'mariadb mysql': ...,
        mssql: ...,
        'snowflake db2': ...,
        ibmi: ...,
        hana: ..., // append to the end
      },
    );
  });
```
But here it looks good to keep the alphabetical order:
```javascript
  it('supports the comment option', () => {
    expectsql(
      () => queryGenerator.createTableQuery('myTable', { myColumn: 'DATE' }, { comment: 'Foo' }),
      {
        default: ...,
        hana: ...,      // keep the alphabetical order
        'mariadb mysql': ...,
        postgres: ...,
        snowflake: ...,
      },
    );
  });
```

Is there any convention on the order of dialects?

**Comment by WikiRik** [comment, marked as answer]
No convention yet. We will not enforce that until we have a formatter rule in place for that.

I personally prefer putting new dialects at the end, but both are fine.

In the case of the createTableQuery unit test file I used the order of the dialects in `test-unit-all` script of core (which is the same order as in the unit test job in CI)
